### DISCUSSION TITLE
Notifications, e-mail and xmpp

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#505] [Q&A] [answered]
Hi,

I wish to use the notifications provided with Apprise.

First I used this for e-mails: 
mailto://{arnauld}:{psswd}@{arnauld.org}:{587}?smtp={arnauld.org}
But didn't receive any e-mails.

Then I tried with xmpp (I have my own home server self hosting e-mails and xmpp (metronome):
xmpp://{arnauld}/{psswd}@{arnauld.org}
But didn't receive anything either.
Do I have to install sleekxmpp on my Birdnet-Pi?

Thank you.

**Comment by ehpersonal38** [comment, marked as answer]
What's the output of pressing the Send Test Notification button?

**Reply by blogdemoi** [reply]
2022-08-18 21:44:07,623 - ERROR - Unparseable URL xmpps://{arnauld}/{xxxx}@{arnauld.org}
2022-08-18 21:44:07,628 - ERROR - Unparseable E-Mail URL mailto://{arnauld}:{xxxx}@{arnauld.org}:{587}?smtp={arnauld.org}
2022-08-18 21:44:07,629 - ERROR - There are no service(s) to notify

**Reply by ehpersonal38** [reply]
The URLs shouldn't have any curly brackets in them, so change
`mailto://{arnauld}:{xxxx}@{arnauld.org}:{587}?smtp={arnauld.org}`

to

`mailto://arnauld:xxxx@arnauld.org:587? smtp=arnauld.org`

:)

**Reply by blogdemoi** [reply]
Thank you.

I tried:

> mailto://arnauld:psswd@arnauld.org:587? smtp=arnauld.org
> mailtos://arnauld:psswd@arnauld.org:587? smtp=arnauld.org
> xmpp://arnauld/psswd@arnauld.org
> xmpps://arnauld/psswd@arnauld.org

But still receive errors:

> 2022-08-19 02:00:00,603 - ERROR - Unparseable URL xmpp://arnauld/k...g
> 2022-08-19 02:00:00,607 - ERROR - Unparseable URL xmpps://arnauld/k...g
> 2022-08-19 02:00:00,617 - INFO - Notifying 1 service(s) asynchronously.
> 2022-08-19 02:00:20,664 - WARNING - A Connection error occurred sending Email notification to arnauld.org.

**Reply by blogdemoi** [reply]
I noticed I have an error in the xmpp, it should be:
xmpp://arnauld:psswd@arnauld.org
xmpps://arnauld:psswd@arnauld.org

But still doesn't work:

> 2022-08-19 02:19:30,280 - ERROR - Unparseable URL xmpp://arnauld:xxx@arnauld.org
> 2022-08-19 02:19:30,282 - ERROR - Unparseable URL xmpps://arnauld:xxx@arnauld.org
> 2022-08-19 02:19:30,283 - ERROR - There are no service(s) to notify

**Reply by ehpersonal38** [reply]
Also you do need to install sleekxmpp, it seems 

https://github.com/caronc/apprise/wiki/Notify_xmpp

**Reply by blogdemoi** [reply]
I also tried using gmail:

> mailto://arnauld:psswd@gmail.com

Unfortunately it doesn't work. Maybe there are some ports to open somewhere, but where?

**Reply by ehpersonal38** [reply]
Read this for Gmail: https://github.com/mcguirepr89/BirdNET-Pi/discussions/337#discussioncomment-2855550

**Reply by blogdemoi** [reply]
Thank you. No luck, still have the same error:

> 2022-08-19 19:48:47,193 - INFO - Applying Google Mail Defaults
> 2022-08-19 19:48:47,201 - INFO - Notifying 1 service(s) asynchronously.
> 2022-08-19 19:49:07,247 - WARNING - A Connection error occurred sending Email notification to smtp.gmail.com.

**Reply by ehpersonal38** [reply]
That error is a generic "invalid credentials" error (I can replicate it by changing one letter of my password to something random). So you can reach Google's servers fine, but your credentials are probably invalid. 

So for `arnauld@gmail.com`, your service URL needs to look like:
`mailto://arnauld:hsdjfaisfnmsxios@gmail.com`

Is the password you're using the 16-digit long app password?

**Reply by blogdemoi** [reply]
"So for arnauld@gmail.com, your service URL needs to look like:
mailto://arnauld:hsdjfaisfnmsxios@gmail.com
Is the password you're using the 16-digit long app password?"

Thank you very much for your patience. You give me hope again :-) I received the notification in gmail. Now I know it is not related to my Birdnet system I will try to make it work  for my e-mail xxx@arnauld.org and my xmpp server.

**Reply by ehpersonal38** [reply]
You're welcome :)

SMTP can be a pain... You may find this tool or one similar handy: https://www.gmass.co/smtp-test

You can mess with the input parameters and hopefully identify the right credentials and ports and whatnot for your SMTP server. Then you can work on getting Apprise to work with those. Good luck!

**Reply by blogdemoi** [reply]
Well, it worked... Now it doesn't work anymore and I didn't touch anything.

**Reply by ehpersonal38** [reply]
Any logs from test notification?

**Reply by blogdemoi** [reply]
It is working again, I think it is related to my e-mail server which is also on a Raspberry Pi (3B) and sometimes it may be slow due to the amount of work on it (there is also a cloud on it and a blog,...). Sorry for this, could you please close the discussion as I consider it as solved? I give up on the xmpp part, e-mail notifications is enough for me.

Thank you again.
